src/OrchardCore.Modules/OrchardCore.ContentFields/ContentItemDisplayExtensions.cs

```diff
@@ -0,0 +1,39 @@
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Localization;
+using OrchardCore.ContentLocalization;
+using OrchardCore.ContentManagement;
+using OrchardCore.ContentManagement.Routing;
+
+namespace OrchardCore.ContentFields
+{
+    public static class ContentItemDisplayExtensions
+    {
+        public static async Task<string> GetContentItemRouteAndCulture(this ContentItem contentItem, IContentManager contentManager, IStringLocalizer S)
+        {
+            var displayText = string.Empty;
+            var additionalParts = new List<string>();
```

Comment:
this would be a `StringBuilder` from the string builder pool

Reply:
moved and reflected in [ContentPickerContentHandler.cs]